src/orquestra/sdk/_base/cli/_dorq/_workflow/_logs.py

```diff
@@ -62,23 +69,53 @@ def _on_cmd_call_with_exceptions(
         wf_run_id: t.Optional[WorkflowRunId],
         config: t.Optional[ConfigName],
         download_dir: t.Optional[Path],
+        task: t.Optional[bool],
+        system: t.Optional[bool],
+        env_setup: t.Optional[bool],
     ):
         # The order of resolving config and run ID is important. It dictates the flow
         # user sees, and possible choices in the prompts.
         resolved_config = self._config_resolver.resolve(wf_run_id, config)
         resolved_wf_run_id = self._wf_run_resolver.resolve_id(
             wf_run_id, resolved_config
         )
+
+        # Get the available logs
         logs = self._wf_run_repo.get_wf_logs(
             wf_run_id=resolved_wf_run_id, config_name=resolved_config
         )
 
-        if download_dir is not None:
-            dump_path = self._dumper.dump(
-                logs=logs,
-                wf_run_id=resolved_wf_run_id,
-                dir_path=download_dir,
-            )
-            self._presenter.show_dumped_wf_logs(dump_path)
-        else:
-            self._presenter.show_logs(logs)
+        # Resolve the log type switches. This must happen after getting the logs as we
+        # need to check against which logs are available.
+        resolved_task_switch: bool
+        resolved_system_switch: bool
+        resolved_env_setup_switch: bool
+        (
+            resolved_task_switch,
+            resolved_system_switch,
+            resolved_env_setup_switch,
+        ) = self._wf_run_resolver.resolve_log_switches(task, system, env_setup, logs)
+
+        for switch, log, identifier in zip(
+            [resolved_task_switch, resolved_system_switch, resolved_env_setup_switch],
+            [logs.per_task, logs.system, logs.env_setup],
+            ["per task", "system", "env setup"],
```

Comment:
**suggestion**: please use an `Enum` instead of magic strings. It would enable using static typecheckers to our advantage.

Reply:
👍